### ISSUE TITLE
Class __init__ "Too few arguments" should be a better error message

### ISSUE THREAD

**Issue by indigoviolet** [python/mypy#9340]
Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request?

Feature request

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
class Foo:
    def __init__(self, bar: int):
        pass


f = Foo()
```

* What is the actual behavior/output?

```
➜ python test.py
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    f = Foo()
TypeError: __init__() missing 1 required positional argument: 'bar'

~ via ⬢ 13.8.0 via 🐍 3.8.1 
➜ mypy --pretty test.py
test.py:6: error: Too few arguments for "Foo"
    f = Foo()
        ^
Found 1 error in 1 file (checked 1 source file)
```

* What is the behavior/output you expect?

mypy should say which arguments are missing, like Python does. This was fixed for functions in #505, but it doesn't seem to work for class __init__.

* What are the versions of mypy and Python you are using?

0.782, 3.8.1

*  Do you see the same issue after installing mypy from Git master?

I have not tested this.

* What are the mypy flags you are using? (For example --strict-optional)

--pretty

* If mypy crashed with a traceback, please paste
  the full traceback below.

(You can freely edit this text, please remove all the lines
you believe are unnecessary.)

**Comment by abhisht51** [comment]
Hi, 
This would be my first time contributing to this repo. Can I take this up ?

**Comment by gvanrossum** [comment]
Sure, give it a try. Read CONTRIBUTING.md to get started.

**Comment by Shkurupii** [comment]
I took a look at this issue recently. The problem is reproduced not only on "init" but for empty function as well.
With next samples:
```python
a = 2
a.__pow__()


class Foo:
    def __init__(self, bar: int):
        pass


c = Foo()


def foo(baz: int, bas: int):
    pass


foo()

```

mypy returns:
```text
(mypy) ➜  mypy git:(master) ✗ mypy samples-9340.py
samples-9340.py:2: error: Too few arguments for "__pow__" of "int"
samples-9340.py:10: error: Too few arguments for "Foo"
samples-9340.py:17: error: Too few arguments for "foo"
Found 3 errors in 1 file (checked 1 source file)
```
As I understand, the function doesn't enter in that main "if" because argument_names is [], subsequently, the length is less then 1, "all" returns False as well. My proposal https://github.com/Shkurupii/mypy/commit/0c759bcdb5c569242aeb4a56e20d7743fcf5a520 is to leave a check only for argument_names is None only. The mypy in this way returns errors similar to the python interpreter.
Basically, the idea is to treat everything in "Missing positional argument" way no matter zero arguments passed or more.
```text
(mypy) ➜  mypy git:(issue-9340) ✗ mypy samples-9340.py 
samples-9340.py:2: error: Missing positional argument
samples-9340.py:10: error: Missing positional argument "bar" in call to "Foo"
samples-9340.py:17: error: Missing positional arguments "baz", "bas" in call to "foo"
Found 3 errors in 1 file (checked 1 source file)
```
But in that case, it requires a lot of refactoring over tests, 197 tests are broken.
@gvanrossum what do you think? give me a few words of advice

**Comment by gvanrossum** [comment]
I'm sorry, I don't have time right now.

**Comment by Shkurupii** [comment]
No problem.
@JukkaL maybe you suggest smth.

**Comment by Shkurupii** [comment]
@ilevkivskyi, maybe you can suggest

**Comment by gvanrossum** [comment]
Maybe try a simpler issue rather than spamming all the devs.

**Comment by PrasanthChettri** [comment]
@gvanrossum  can I take this one ?

**Comment by gvanrossum** [comment]
@PrasanthChettri Give it a try!

**Comment by abhinaypandey02** [comment]
@gvanrossum sorry I am new, I created a pull request without asking for permission. please consider it!

**Comment by abhinaypandey02** [comment]
@gvanrossum I am sorry I forgot to ask for permission. Please consider my work on this issue. It's my first time contributing.

**Comment by gvanrossum** [comment]
No worries, someone else is already reviewing your PR.
